source/Mlos.Python/mlos/OptimizerEvaluationTools/unit_tests/TestOptimizerEvaluator.py

```diff
@@ -0,0 +1,141 @@
+        self.assertTrue(optimizer_evaluator_config.include_execution_trace_in_report)
+
+        optimizer_config = bayesian_optimizer_config_store.default
+        objective_function_config = objective_function_config_store.default
+
+        print(optimizer_evaluator_config.to_json(indent=2))
+        print(optimizer_config.to_json(indent=2))
+        print(objective_function_config.to_json(indent=2))
+
+        optimizer_evaluator = OptimizerEvaluator(
+            optimizer_evaluator_config=optimizer_evaluator_config,
+            objective_function_config=objective_function_config,
+            optimizer_config=optimizer_config
+        )
+
+        optimizer_evaluation_report = optimizer_evaluator.evaluate_optimizer()
+
+        with pd.option_context('display.max_columns', 100):
+            print(optimizer_evaluation_report.regression_model_goodness_of_fit_state.get_goodness_of_fit_dataframe(DataSetType.TRAIN).tail())
+            for optimum_name, optimum_over_time in optimizer_evaluation_report.optima_over_time.items():
+                print("#####################################################################################################")
+                print(optimum_name)
+                print(optimum_over_time.get_dataframe().tail(10))
+                print("#####################################################################################################")
+
+        optimizer_evaluation_report.write_to_disk(target_folder=self.temp_dir)
+
+        # Now let's do it again with the unpickled optimizer.
+        #
+        unpickled_objective_function = pickle.loads(optimizer_evaluation_report.pickled_objective_function_initial_state)
+        unpickled_optimizer = pickle.loads(optimizer_evaluation_report.pickled_optimizer_initial_state)
+        optimizer_evaluator_2 = OptimizerEvaluator(
+            optimizer_evaluator_config=optimizer_evaluator_config,
+            objective_function=unpickled_objective_function,
+            optimizer=unpickled_optimizer
+        )
+
+        optimizer_evaluation_report_2 = optimizer_evaluator_2.evaluate_optimizer()
+        with pd.option_context('display.max_columns', 100):
+            print(optimizer_evaluation_report_2.regression_model_goodness_of_fit_state.get_goodness_of_fit_dataframe(DataSetType.TRAIN).tail())
+            for optimum_name, optimum_over_time in optimizer_evaluation_report_2.optima_over_time.items():
+                print("#####################################################################################################")
+                print(optimum_name)
+                print(optimum_over_time.get_dataframe().tail(10))
+                print("#####################################################################################################")
+
+
+    def test_named_configs(self):
+        """Tests all named optimizer configurations against all named objective functions."""

```

Comment:
The cross product is actually too expensive to test, but we can test a bunch of pairs.